app/main.py

```diff
@@ -1,6 +1,15 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    a = []
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> any:
+        result = func(*args, **kwargs)
+        if result not in a:
+            a.append(result)
```

Comment:
Appending only the result to the cache list `a` is insufficient for a proper caching mechanism. You need to store the result associated with its input arguments, typically using a dictionary where the key is the arguments and the value is the result.